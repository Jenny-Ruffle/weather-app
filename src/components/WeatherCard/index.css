.weather-card {
    scroll-snap-align: start;
    flex-shrink: 0;
    width: 170px;
    position: relative;
    background-color: white;
    border: 1px solid white;
    border-radius: 5px;
    color: gray;
    margin: 0 20px;
    flex: 0 0 170px;
    display: block;
}

.weather-card_prefix {
    font-family: 'Open Sans', sans-serif;
    font-size: 16px;
    color: black;
    margin: 12px 0;
}

.weather-card_temp {
    font-family: 'Jost', sans-serif;
    font-size: 36px;
    color: #0c0c4e;
    margin: 12px 0;
}