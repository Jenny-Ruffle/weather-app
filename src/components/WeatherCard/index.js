import React from 'react'
import './index.css'
import WeatherTypeIcon from '../WeatherTypeIcon'

const WeatherCard = ({data}) => {
    const {day, date, temp, type, high, low} = data

    return (
        <div className='weather-card'>
            <p className='weather-card_prefix'>{date}</p>
            <p className='weather-card_prefix'>{day}</p>
            <h1 className='weather-card_temp'>{temp}°c</h1>
            <WeatherTypeIcon type={type} />
            <p className='weather-card_prefix'>High: {high}°c | Low: {low}°c</p>
        </div>
    )
}

export default WeatherCard