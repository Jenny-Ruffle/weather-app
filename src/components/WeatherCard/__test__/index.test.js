import React from 'react';
import TestRenderer from 'react-test-renderer';
import WeatherCard from '../index.js';

const mockData = {
    day: 'Monday',
    date: '11/05',
    temp: '13',
    type: 'cloudy',
    high: '14',
    low: '8'
}

describe('Weather card', () => {
    it('matches the snapshot', () => {
        const newSnapshot = TestRenderer.create(<WeatherCard data={mockData} />).toJSON();
        expect(newSnapshot).toMatchSnapshot();
    });
  });