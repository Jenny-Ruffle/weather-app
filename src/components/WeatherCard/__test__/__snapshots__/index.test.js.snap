// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather card matches the snapshot 1`] = `
<div
  className="weather-card"
>
  <p
    className="weather-card_prefix"
  >
    11/05
  </p>
  <p
    className="weather-card_prefix"
  >
    Monday
  </p>
  <h1
    className="weather-card_temp"
  >
    13
    °c
  </h1>
  <img
    alt="weather-svg"
    src="cloudy.svg"
  />
  <p
    className="weather-card_prefix"
  >
    High: 
    14
    °c | Low: 
    8
    °c
  </p>
</div>
`;
